### PR TITLE
[Draft]Add Multimodal RAG notebook

### PR DESCRIPTION
**Pull request by openvino-dev-samples**
image

image

### CODE REVIEW

File: notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb [hunk 1]
```diff
@@ -0,0 +1,1054 @@
+{
```
Comment (eaidova) [inline notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.        convert_phi3_model(vlm_model_id, vlm_model_path, compression_configuration)
</pre><p><a href="https://github.com/huggingface/optimum-intel/pull/977" target="_blank">https://github.com/huggingface/optimum-intel/pull/977</a> phi3-vision is planned to be integrated to optimum-intel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1830717253'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='9'/>

Comment (openvino-dev-samples) [reply notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:1]: Done, replace with optimum-intel

File: notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb [hunk 2]
```diff
@@ -0,0 +1,1168 @@
...
```
Comment (eaidova) [inline notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.        "transformers==4.45" \
</pre><p>what the reason for pinning transforemrs version?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1846060949'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='2'/>

**Comment by eaidova** [comment]
@openvino-dev-samples for me everything looks good, thanks.

Couple of comments:
Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)

Is there any plans to integrate OV Visual Language models directly in llama-index?

**Comment by openvino-dev-samples** [comment]
> @openvino-dev-samples for me everything looks good, thanks.
> 
> Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> 
> Is there any plans to integrate OV Visual Language models directly in llama-index?

Thanks for your review, the integration is already done in llama-index 

https://docs.llamaindex.ai/en/stable/examples/multi_modal/openvino_multimodal/

BTW is there an example for phi3-vision's accuracy aware quantization ?

**Comment by nikita-savelyevv** [comment]
> Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)


I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.

Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.

File: notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb (hunk 2 above)
Comment (aleksandr-mokrov) [inline notebooks/multimodal-rag/multimodal-rag-llamaindex.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    demo = make_demo(filepath, build_index, search, generate, load_image, stop)
</pre><p>There are no <code>load_image </code>argument in <code>make_demo </code>function.﻿</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2497/discussion/#comment-1846792334'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='40'/>

**Comment by openvino-dev-samples** [comment]
> > Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> 
> I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.
> 
> Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.

Hi, as my test, the accuracy with this configuration is not satisfied:
`optimum-cli export openvino --model {vlm_model_id}  {vlm_model_path} --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32`

**Comment by nikita-savelyevv** [comment]
> > > Couple of comments: Possibly it is better to move on accuracy aware quantization for VLM using optimum-cli, need to provide `--weight-format int4 --dataset contextual` options for that (fyi @nikita-savelyevv)
> > 
> > 
> > I'll add that an algorithm itself needs to be specified, e.g. `--weight-format int4 --dataset contextual --awq`.
> > Also, the default number of samples of 128 might be too large, so it can be reduced with `--num-samples 32`.
> 
> Hi, as my test, the accuracy with this configuration is not satisfied: `optimum-cli export openvino --model {vlm_model_id} {vlm_model_path} --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32`

Thanks for the information! Have you compared it against the configuration below?
```
compression_config = {
    "mode": nncf.CompressWeightsMode.INT4_SYM,
    "group_size": 64,
    "ratio": 0.6,
}
```

Yes, this configuration brings more reasonable responses compared to optimum-cli

**Comment by openvino-dev-samples** [comment]
hi @eaidova looks the CI is out of resource to validate this notebook, any suggestions ? thanks

**Comment by eaidova** [comment]
@openvino-dev-samples I can suggest to try internvl2-1b-instruct or nano-llava for ci testing, it is small enough to fit into precommit

**Comment by openvino-dev-samples** [comment]
> --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32

Sorry, I made a mistake before. The result looks good with this accuracy aware config now

**Comment by openvino-dev-samples** [comment]
> > --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32
> 
> Sorry, I made a mistake before. The result looks good with this accuracy aware config now

However I find it impossible to run this quantization method in a client PC with 32GB RAM

**Comment by nikita-savelyevv** [comment]
> > > --trust-remote-code --weight-format int4 --dataset contextual --awq --num-samples 32
> > 
> > 
> > Sorry, I made a mistake before. The result looks good with this accuracy aware config now
> 
> However I find it impossible to run this quantization method in a client PC with 32GB RAM

Could you please verify that NNCF 2.14 is installed? It was released recently and it contains significant improvements in terms of peak RAM during data-aware compression.

**Comment by brmarkus** [comment]
> Could you please verify that NNCF 2.14 is installed? It was released recently and it contains significant improvements in terms of peak RAM during data-aware compression.

At least this pull-request doesn't touch "requirements.txt" and doesn't check to make sure a recent version of NNCF is present. On my 64GB laptop the max. system memory is shortly fully used.

**Review by eaidova — APPROVED** [review]

**Pull request merged**